app/code/Transoft/Blog/ViewModel/Bloglist.php

```diff
@@ -0,0 +1,81 @@
+<?php
+declare(strict_types=1);
+namespace Transoft\Blog\ViewModel;
+
+use Magento\Catalog\Block\Product\View;
+use Magento\Framework\View\Element\Block\ArgumentInterface;
+use Magento\Framework\App\ObjectManager;
+use Transoft\Blog\Helper\Data;
+use Transoft\Blog\Model\ResourceModel\Post\Collection;
+use Transoft\Blog\Model\ResourceModel\Post\CollectionFactory as PostCollectionFactory;
+
+/**
+ * Class Bloglist
+ *
+ * Blog list view model class
+ */
+class Bloglist implements ArgumentInterface
+{
+    /**
+     * @var PostCollectionFactory
+     */
+    protected $modelFactory;
+
+    /**
+     * Transoft\Blog\Helper\Data $helper
+     */
+    protected $helper;
+
+    /**
+     * Constructor.
+     *
+     * @param PostCollectionFactory $modelFactory
+     * @param Data $helper
+     */
+    public function __construct(
+        PostCollectionFactory $modelFactory,
+        Data $helper
+    ) {
+        $this->modelFactory = $modelFactory;
+        $this->helper = $helper;
+    }
+
+    /**
+     * Returns model collection
+     *
+     * @return Collection
+     */
+    public function retrieveModel()
```

Comment:
Please provide a meaningful method name. It should reflect what it does

Reply:
fixed